packages/hardhat-create2-proxy/docs/decisions/001-deterministic-secondary-key-derivation-for-create2-proxy-deployment.md

```diff
@@ -0,0 +1,72 @@
+# ADR: Deterministic Secondary Key Derivation for CREATE2 Proxy Deployment
+
+## Status
+
+Accepted
+
+## Date: 19-05-2025
+
+## Context
+
+In order to ensure **consistent contract addresses across multiple EVM chains** using `CREATE2`, we must control the **deployer address**, since it is a critical part of the formula used by the EVM to derive the final contract address.
```

Comment:
The derived key is only used for the deployment of the Create2Factory, right? That does affect the address of a contract that is deployed using that factory. However, the derived key isn’t used for contract deployment directly. It’s not used by deployUsingCreate2. So I think this one needs to be clear about the fact that this is only related to the Create2Factory deployment.